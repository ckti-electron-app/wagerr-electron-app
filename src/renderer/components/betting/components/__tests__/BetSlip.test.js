import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import BetSlip from '../BetSlip.vue';

jest.mock('@/services/api/wagerrRPC');

const localVue = createLocalVue();

localVue.use(Vuex);

describe('component: BetSlip', () => {
  let store;
  let getters;
  let actions;

  const testBets = [
    {
      betId: '1',
      outcome: 1,
      winner: 'Levante',
      odds: 26200,
      eventDetails: {
        event_id: 63200,
        sport: 'Soccer',
        tournament: 'LaLiga',
        starting: 1581163200,
        tester: 2238,
        teams: { home: 'Levante', away: 'Leganés' },
        odds: [
          { mlHome: 26200, mlAway: 31900, mlDraw: 33600 },
          { spreadPoints: 0, spreadHome: 0, spreadAway: 0 },
          { totalsPoints: 25, totalsOver: 23400, totalsUnder: 17400 }
        ]
      },
      betType: null,
      handicap: null,
      totalValue: null,
      availability: true
    },
    {
      betId: '2',
      outcome: 1,
      winner: 'Göztepe',
      odds: 18400,
      eventDetails: {
        event_id: 63374,
        sport: 'Soccer',
        tournament: 'Super Lig',
        starting: 1581354000,
        tester: 1435,
        teams: { home: 'Göztepe', away: 'Çaykur Rizespor' },
        odds: [
          { mlHome: 18400, mlAway: 50000, mlDraw: 40000 },
          { spreadPoints: 0, spreadHome: 0, spreadAway: 0 },
          { totalsPoints: 25, totalsOver: 20000, totalsUnder: 20000 }
        ]
      },
      betType: null,
      handicap: null,
      totalValue: null,
      availability: true
    },
    {
      betId: '3',
      outcome: 7,
      winner: 'Göztepe',
      odds: 20000,
      eventDetails: {
        event_id: 63374,
        sport: 'Soccer',
        tournament: 'Super Lig',
        starting: 1581354000,
        tester: 1435,
        teams: { home: 'Göztepe', away: 'Çaykur Rizespor' },
        odds: [
          { mlHome: 18400, mlAway: 50000, mlDraw: 40000 },
          { spreadPoints: 0, spreadHome: 0, spreadAway: 0 },
          { totalsPoints: 25, totalsOver: 20000, totalsUnder: 20000 }
        ]
      },
      betType: 'total',
      handicap: null,
      totalValue: 'Under2.5',
      availability: true
    }
  ];

  beforeEach(() => {
    getters = {
      balance: () => 10000,
      immature: () => 0,
      pending: () => 0,
      betSlip: () => testBets,
      getNumBets: () => 2,
      getNetworkType: () => 'Testnet',
      convertOdds: () => () => 1.5,
      betType: () => 'multi'
    };

    actions = {
      removeBetFromSlip: jest.fn(),
      clearBetSlip: jest.fn(),
      setBetType: jest.fn()
    };

    store = new Vuex.Store({
      getters,
      actions
    });
  });

  it('should render component', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });
    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('should match snapshot', () => {
    expect.hasAssertions();
    const wrapper = mount(BetSlip, {
      store,
      localVue
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should display correct input value', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });

    const input = wrapper.find('input');
    input.setValue('125');

    expect(input.element.value).toBe('125');
    expect(wrapper.vm.multiBetNumber).toBe(125);
  });

  it('should calculate correct multiBetNumber value', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });

    expect(wrapper.vm.multiBetNumber).toBe(0);

    const input = wrapper.find('input');
    input.setValue('125');

    expect(wrapper.vm.multiBetNumber).toBe(125);
  });

  it('should calculate correct multiPotentialWinnings value', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });
    expect(wrapper.vm.multiPotentialWinnings).toBe('0.00000000');

    const input = wrapper.find('input');
    input.setValue('125');

    expect(wrapper.vm.multiPotentialWinnings).toBe('1140.38800000');
  });

  it('should calculate correct wagerrCode', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });
    expect(wrapper.vm.wagerrCode).toBe(' tWGR');
  });

  it('should execute placeBet with correct params', () => {
    expect.hasAssertions();
    const wrapper = shallowMount(BetSlip, {
      store,
      localVue
    });
    const placeBetMock = jest.fn().mockResolvedValue(43);
    wrapper.setMethods({ placeBet: placeBetMock });

    const input = wrapper.find('input');
    input.setValue('125');

    wrapper
      .find('div.bet-slip__multi-summary')
      .find('button')
      .trigger('click');

    expect(placeBetMock).toHaveBeenCalledWith(
      null,
      [
        { eventId: 63200, outcome: 1 },
        { eventId: 63374, outcome: 1 },
        { eventId: 63374, outcome: 7 }
      ],
      125
    );
  });
});
