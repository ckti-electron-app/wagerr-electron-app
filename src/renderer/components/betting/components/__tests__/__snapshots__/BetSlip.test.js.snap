// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: BetSlip should match snapshot 1`] = `
<div
  class="bet-slip"
  id="bet-slip"
>
  <h4>
    <span
      class="bet-slip__bet-type-btn"
    >

      Single

    </span>

    |

    <span
      class="bet-slip__bet-type-btn bet-slip__bet-type-btn--active"
    >

      Multi

    </span>

    <button
      class="btn pull-right waves-effect waves-light"
    >

      Clear Slip

    </button>
  </h4>

  <div
    class="bet-list-scroll"
  >
    <div
      class="bet-list"
    >
      <ul>
        <li
          class="bet-card"
        >
          <div
            class="bet-card__header"
          >
            <span
              class="bet-card__header-text"
            >

      Levante vs
      Leganés

            </span>

            <div
              class="bet-card__clear-btn"
            >
              <i>
                ×
              </i>
            </div>
          </div>

          <div
            class="bet-card__selection"
          >
            <div
              class="bet-card__selection-title"
            >
              Your Pick:
            </div>

            <span
              class="bet-card__selection-winner"
            >

      Levante

            </span>

            <!---->

            <!---->

            <span
              class="bet-card__selection-odds"
            >

      1.5

            </span>
          </div>

          <!---->
        </li>
        <li
          class="bet-card"
        >
          <div
            class="bet-card__header"
          >
            <span
              class="bet-card__header-text"
            >

      Göztepe vs
      Çaykur Rizespor

            </span>

            <div
              class="bet-card__clear-btn"
            >
              <i>
                ×
              </i>
            </div>
          </div>

          <div
            class="bet-card__selection"
          >
            <div
              class="bet-card__selection-title"
            >
              Your Pick:
            </div>

            <span
              class="bet-card__selection-winner"
            >

      Göztepe

            </span>

            <!---->

            <!---->

            <span
              class="bet-card__selection-odds"
            >

      1.5

            </span>
          </div>

          <!---->
        </li>
        <li
          class="bet-card"
        >
          <div
            class="bet-card__header"
          >
            <span
              class="bet-card__header-text"
            >

      Göztepe vs
      Çaykur Rizespor

            </span>

            <div
              class="bet-card__clear-btn"
            >
              <i>
                ×
              </i>
            </div>
          </div>

          <div
            class="bet-card__selection"
          >
            <div
              class="bet-card__selection-title"
            >
              Your Pick:
            </div>

            <!---->

            <span
              class="bet-card__selection-odds"
            >

      Under2.5

            </span>

            <!---->

            <span
              class="bet-card__selection-odds"
            >

      1.5

            </span>
          </div>

          <!---->
        </li>
      </ul>
    </div>
  </div>

  <div
    class="bet-slip__multi-summary"
  >
    <div>
      Total Legs: 2
    </div>

    <div
      class="input-field"
    >
      <div
        class="bet-slip__multi-summary-bet"
      >
        <label
          class="bet-slip__multi-summary-bet-label"
          for="multi-bet"
        >

          Bet

        </label>

        <input
          class="bet-stake validate"
          id="multi-bet"
          maxlength="10"
          name="Bet Id"
          placeholder="Enter Bet"
          type="text"
        />
      </div>
    </div>

    <div
      class="bet-slip__warning bet-slip__warning--light display-none"
    >



    </div>

    <div
      class="bet-slip__multi-summary-return"
    >

      To Win: 0.00000000  tWGR

    </div>

    <button
      class="btn waves-effect waves-light disabled"
    >

      Place Bet

    </button>
  </div>
</div>
`;
